provisioning/device/samples/getting started/X509Sample/ProvisioningDeviceClientSample.cs

```diff
@@ -44,7 +44,7 @@ public async Task RunSampleAsync()
             DeviceRegistrationResult result = await provClient.RegisterAsync();
 
             Console.WriteLine($"Registration status: {result.Status}.");
-            if (result.Status != ProvisioningRegistrationStatusType.Assigned)
+            if (result.Status != ProvisioningRegistrationStatus.Assigned)
```

Comment:
Now deserializing to the enum instead of a string, and requiring string comparisons. Yay!